packages/@aws-cdk/cloudformation-include/test/valid-templates.test.ts

```diff
@@ -444,10 +444,64 @@ describe('CDK Include', () => {
     });
   });
 
-  test("throws an exception when encountering a Resource type it doesn't recognize", () => {
+  test('can include a template with a custom resource that only defines a type', () => {
+    includeTestTemplate(stack, 'non-existent-resource-type.json');
+    expect(stack).toHaveResourceLike('AWS::MyService::Custom');
+  });
+
+  test('can include a template with a custom resource that has custom properties', () => {
+    includeTestTemplate(stack, 'custom-resource-properties.json');
+    expect(stack).toHaveResourceLike('AWS::MyService::Custom', {
+      "CustomProp": "CustomValue",
+    });
+  });
+
+  test('can include a template with a custom resource that uses cloudformation functions', () => {
+    includeTestTemplate(stack, 'custom-resource-with-functions.json');
+    expect(stack).toHaveResourceLike('AWS::MyService::Custom', {
+      "CustomStringProp": {
+        "Ref": "AWS::NoValue",
+      },
+    });
+  });
+
+  test('can include a template with a custom resource that uses attributes', () => {
+    includeTestTemplate(stack, 'custom-resource-with-attributes.json');
+    expect(stack).toHaveResourceLike('AWS::MyService::Custom', {
+      "Metadata": {
+        "Object1": "Value1",
+        "Object2": "Value2",
+      },
+      "CreationPolicy": {
+        "AutoScalingCreationPolicy": {
+          "MinSuccessfulInstancesPercent": 90,
+        },
+      },
+      "DeletionPolicy": "Retain",
+      "DependsOn": [ "CustomResource" ],
+    }, ResourcePart.CompleteDefinition);
+
+    expect(stack).toHaveResourceLike('AWS::MyService::AnotherCustom', {
+      "UpdatePolicy": {
+        "AutoScalingReplacingUpdate": {
+          "WillReplace": "false",
+        },
+      },
+      "UpdateReplacePolicy": "Retain",
+    }, ResourcePart.CompleteDefinition);
+  });
```

Comment:
Actually, shouldn't this just check that the resulting template is the same as the input one?